templateflow/templateflow_get.py

```diff
@@ -0,0 +1,197 @@
+
+    i_group.add_argument('-m', '--metadata', 
+		    action='store_true',
+		    default=False,
+		    help='Get full metadata for given template')
+
+    i_group.add_argument('-C', '--citations', 
+		    action='store_true',
+		    default=False,
+		    help='Get template citations')
+
+    i_group.add_argument('-B', '--bibtex-citations', 
+		    action='store_true',
+		    default=False,
+		    help='Get template citation in bibtex-citations')
+
+    m_group = parser.add_argument_group('misc options')
+    m_group.add_argument('-o', '--tf-cache', 
+		    action='store',
+		    default='unset',
+		    help='Directory to save templates in. default `${HOME}/.cache/templateflow`')
+
+    m_group.add_argument('-v', '--version', 
+		    action='store_true',
+		    default=False, 
+		    help='Display version of this tools AND TemplateFlow Client version')
+
+    parser.add_argument('args', nargs='*',metavar=('template [template ...]') )
+    opts = parser.parse_args(sys.argv[1:])
+    args = opts.args
+    return opts, args
+
+
+def version_info():
+    import templateflow
+    prog = os.path.basename(sys.argv[0])
+    print(f'{prog} v{TOOL_VERSION}')
+    print(f'Using TemplateFlow Client version {templateflow.__version__}')
+    sys.exit(0)
+
+def templates_info():
+    import templateflow.api as tf
+    templates=tf.templates()
+    ident_fmt = '\n'.join(templates)
+    print(f"{len(templates)} templates:\n-------------------\n{ident_fmt}")
+    sys.exit(0)
+
+def metadata_info(args):    
+    import templateflow.api as tf
+    pp = pprint.PrettyPrinter(indent=4)
+    for tmpl in args:
+        infos=tf.get_metadata(tmpl)
+        pp.pprint(infos)
+    sys.exit(0) 
+
+def citation_info(args, bibtex=True):
+    import templateflow.api as tf
+#    bibtex_format = True if  opts.bibtex_citations else False
+    pp = pprint.PrettyPrinter(indent=4)
+    for tmpl in args:
+        print(tmpl)
+        
+        infos=tf.get_citations(tmpl, bibtex=bibtex)
+        pp.pprint(infos)
+    sys.exit(0) 
+
+def template_retriever(args, options):
+    import templateflow
+    import templateflow.api as tf
+
+    for tmpl in args:
+        try: 
+            tmpl = tf.get(tmpl, raise_empty=True, **options) 
+            print('\n'.join(map(str, tmpl)))
+        except Exception as msg:
+            print(tmpl, msg)
+    sys.exit(0)
+
+
+def main():
+    
+    opts, args = parse_args()
+
+    if opts.version:
+        version_info()
+
+    if opts.list:
+        templates_info()
+        
+    if opts.metadata:
+        metadata_info(args)
+
+    if opts.citations or opts.bibtex_citations:
+        citation_info(args, opts.bibtex_citations)
+
+    if opts.tf_cache != 'unset':
+        #---- TemplatFlow cache dir was specified, export it
+        print("saving to ---------->", opts.tf_cache)
+        print("saving to ---------->", os.path.abspath(opts.tf_cache))
+        os.environ['TEMPLATEFLOW_HOME'] = os.path.abspath(opts.tf_cache)
```

Comment:
this should be managed by templateflow

Reply:
seems to me I was lazy and copied this spnippet from the docs or examples .